src/workerd/api/actor-state.c++

```diff
@@ -919,16 +919,20 @@ v8::Local<v8::Value> deserializeV8Value(
     // counting it against our internal storage error metrics but also throw a KJ exception rather
     // than a jsExceptionThrown error to avoid confusing the normal termination handling code.
     // We don't expect users to ever actually see this error.
-    JSG_REQUIRE(!tryCatch.HasTerminated(),
-        Error, "isolate terminated while deserializing value from Durable Object storage; "
-        "contact us if you're wondering why you're seeing this");
+    auto jsException = tryCatch.Exception();
+    if (!tryCatch.HasTerminated()) {
+      KJ_LOG(ERROR, "actor storage deserialization failed", errorMsg, jsException,
+          key, buf.size(), buf.slice(0, std::min(static_cast<size_t>(3), buf.size())));
```

Comment:
Please leave out at least `buf` if not also `key! This could leak personally identifiable information to logs.

Reply:
Hmm, what about the FAIL_ASSERTs below? I directly copied one of them so should that change too?

Reply:
Yeah, if you would. As long as we have identifying DO info from our log context, we can check the storage backend later.

Reply:
You absolutely sure? This comment indicates there was some thought put into this:
```
  // If we do hit a deserialization error, we log information that will be helpful in understanding
  // the problem but that won't leak too much about the customer's data. We include the key (to help
  // find the data in the database if it hasn't been deleted), the length of the value, and the
  // first three bytes of the value (which is just the v8-internal version header and the tag that
  // indicates the type of the value, but not its contents).
```